azure-pipelines.yml

```diff
@@ -26,25 +27,51 @@ pr:
     include:
     - 'v*'
 
+schedules:
+  - cron: "0 0 * * *"
+    displayName: Daily midnight build
+    branches:
+      include:
+        - master
+    always: false  # Only run if new changes since last scheduled build
+
 jobs:
-  - template: publish.yml@OpenAstronomy
-    parameters:
-
-      # FIXME: we exclude the test_data_out_of_range test since it
-      # currently fails, see https://github.com/astropy/astropy/issues/10409
-      test_command: pytest -p no:warnings --astropy-header -k "not test_data_out_of_range" --pyargs astropy
-      test_extras: test
-
-      # NOTE: for v* tags, we auto-release to PyPI. See
-      # https://openastronomy-azure-pipelines.readthedocs.io/en/latest/publish.html
-      # for information on how to configure things on the Azure Pipelines side
-      ${{ if startsWith(variables['Build.SourceBranch'], 'refs/tags/v') }}:
-        pypi_connection_name : 'pypi_endpoint'
-
-      targets:
-      - sdist
-      - wheels_cp3[678]*linux_i686
-      - wheels_cp3[678]*linux_x86_64
-      - wheels_cp3[678]*macosx_x86_64
-      - wheels_cp3[678]*win32
-      - wheels_cp3[678]*win_amd64
+  # We need to specify a job if the pipeline is triggered.
+  # Azure pipelines will trigger on commits to master, but we don't want to
+  # build wheels for each commit, so we run a noop job on non-scheduled master
+  # builds to save CPU cycles.
+  - ${{ if not(eq(variables['Build.Reason'], 'Schedule')) }}:
+    - job: noop
+      steps:
+        - checkout: none
+
+  # Only run this job on schedule, to build nightly wheels or on a tag to build
+  # release wheels
+  - ${{ if or(startsWith(variables['Build.SourceBranch'], 'refs/tags/v'), eq(variables['Build.Reason'], 'Schedule')) }}:
+    - template: publish.yml@OpenAstronomy
+      parameters:
+
+        # FIXME: we exclude the test_data_out_of_range test since it
+        # currently fails, see https://github.com/astropy/astropy/issues/10409
+        test_command: pytest -p no:warnings --astropy-header -k "not test_data_out_of_range" --pyargs astropy
+        test_extras: test
+
+        # NOTE: for v* tags, we auto-release to PyPI. See
+        # https://openastronomy-azure-pipelines.readthedocs.io/en/latest/publish.html
+        # for information on how to configure things on the Azure Pipelines side
+        ${{ if startsWith(variables['Build.SourceBranch'], 'refs/tags/v') }}:
+          pypi_connection_name : 'pypi_endpoint'
+
+        # Only upload nightly builds on master when the build is scheduled
+        ${{ if and(eq(variables['Build.Reason'], 'Schedule'), eq(variables['Build.SourceBranchName'], 'master')) }}:
+          artifact_project : 'astropy'
+          artifact_feed : 'nightlies'
```

Comment:
Can we call this nightly to make it less cutesy? 😄

Reply:
heh it was originally I changed it because I remade the feed while debugging, I guess I can remake it again.